### PR TITLE
add all docstrings for NumbaIntegrationTestTarget

### LINKED ISSUE #3: Add docstrings to methods in `NumbaIntegrationTestTarget`
Might be helpful to add docstrings to the class `NumbaIntegrationTestTarget`.

### PR DESCRIPTION
**Pull request by esc**
Adding these docsrings should ease the task of adding new targets.

Fixes #3

### CODE REVIEW

**Comment by esc** [comment]
ping @stuartarchibald

File: switchboard.py [hunk 1]
```diff
@@ -135,26 +135,90 @@ def conda_install(env, target):
...
+
+        """
         raise NotImplementedError
 
     @property
     def clone_url(self):
+        """ Canonical clone url for the target.
+
+        This will be used to clone the project if needed. If you omit this, the
+        project will not be cloned and it is assumed, that the target ships
```
Comment (stuartarchibald) [inline switchboard.py]:
```suggestion
        project will not be cloned and it is assumed that the target ships
```

File: switchboard.py [hunk 2]
```diff
@@ -135,26 +135,90 @@ def conda_install(env, target):
...
+            A 'git clone' compatible url.
+
+        """
         raise NotImplementedError
 
     @property
     def target_tag(self):
+        """ The target tag to checkout.
+
+        This function should work out which tag should be checked out and
```
Comment (stuartarchibald) [inline switchboard.py]:
```suggestion
        This function must work out which tag should be checked out and
```

File: switchboard.py [hunk 3]
```diff
@@ -135,26 +135,90 @@ def conda_install(env, target):
...
+
+        """
         raise NotImplementedError
 
     @property
     def conda_dependencies(self):
+        """ Conda dependencies for this project.
+
+        The conda dependencies for this project. If you need to install things
+        in a specific order with multiple, subsequent `conda` calls, use
```
Comment (stuartarchibald) [inline switchboard.py]:
```suggestion
        in a specific order with multiple, subsequent, `conda` calls, use
```

File: switchboard.py [hunk 4]
```diff
@@ -135,26 +135,90 @@ def conda_install(env, target):
...
+        multiple stings. You can include any channel information such as `-c
```
Comment (stuartarchibald) [inline switchboard.py]:
```suggestion
        multiple strings. You can include any channel information such as `-c
```

**Review by stuartarchibald — COMMENTED** [review]
Thanks for the patch. These docstrings are very helpful, few minor comments to resolve then this can be merged.

**Comment by esc** [comment]
@stuartarchibald thanks for the review, changes have been applied. Would suggest to merge when a few of the known 'good' projects go green.

**Review by stuartarchibald — APPROVED** [review]

**Comment by stuartarchibald** [comment]
Thanks for the fix-ups. Merging.

**Pull request merged**